test/unit/org/apache/cassandra/index/SecondaryIndexManagerTest.java

```diff
@@ -56,6 +57,20 @@ public void after()
         TestingIndex.clear();
     }
 
+    @Test
+    public void createSasiAfterSai()
+    {
+        execute(format("CREATE TABLE %s.abc (id int PRIMARY KEY, val text)", KEYSPACE));
+        execute(format("CREATE INDEX ON %s.abc (val) USING 'sai'", KEYSPACE));
+        waitForTableIndexesQueryable(KEYSPACE, "abc");
+        execute(format("INSERT INTO %s.abc (id, val) VALUES (1, 'a')", KEYSPACE));
+        execute(format("SELECT * FROM %s.abc WHERE val = 'a'", KEYSPACE));
+        flush(KEYSPACE);
+        execute(format("CREATE CUSTOM INDEX on %s.abc (val) USING 'org.apache.cassandra.index.sasi.SASIIndex'", KEYSPACE));
+        waitForTableIndexesQueryable(KEYSPACE, "abc");
+        execute(format("SELECT * FROM %s.abc WHERE val = 'a'", KEYSPACE));
```

Comment:
I think this can be simplified to:
```suggestion
        createTable("CREATE TABLE %s (id int PRIMARY KEY, val text)");
        createIndex("CREATE INDEX ON %s (val) USING 'sai'");
        execute("INSERT INTO %s (id, val) VALUES (1, 'a')");
        execute("SELECT * FROM %s WHERE val = 'a'");
        flush();
        createIndex("CREATE INDEX ON %s (val) USING 'org.apache.cassandra.index.sasi.SASIIndex'");
        execute("SELECT * FROM %s WHERE val = 'a'");
```

Reply:
It seems that a minor issue with the test is that it doesn't consistently reproduce the bug. It would only reproduce it with the old code if the concurrent map containing the indexes happens to return the SAI index first. This doesn't seem deterministic, and in fact, the test sometimes passes without the fix. This is not a big deal IMO, because the test aims at a very specific bug that doesn't exist anymore.